scripts/om2files.py

```diff
@@ -0,0 +1,567 @@
+    :return: a list of matching parts
+    """
+    regex = re.compile(regex_str)
+
+    if isinstance(js_obj, list) or isinstance(js_obj, dict):
+        return [i for i in search_json_component(js_obj, regex) if i is not None]
+    else:
+        raise InvalidOM
+
+
+def search_json_field(js_obj: JsonType, field: str,
+                      regex: re.Pattern) -> t.Generator:
+    """
+
+    :param js_obj: A dict or list from a parsed json file
+    :param field: The name of the field to be found
+    :param regex: A compiled regular expression to search values
+    :return: a generator that will produce matching dicts.
+    """
+    def check_match(value: t.Union[list, str]):
+        if isinstance(value, str):
+            if regex.match(value):
+                return True
+        elif isinstance(value, list):
+            for val in value:
+                if isinstance(val, str) and regex.match(val):
+                    return True
+        return False
+
+    if isinstance(js_obj, dict):
+        if field in js_obj and check_match(js_obj[field]):
+            yield js_obj
+        else:
+            for v in search_json_field(list(js_obj.values()), field, regex):
+                if v is not None:
+                    yield v
+    elif isinstance(js_obj, list):
+        for i in js_obj:
+            for j in search_json_field(i, field, regex):
+                if j is not None:
+                    yield j
+
+
+def find_json_field_name(js_obj: JsonType, field: str, regex_str: str):
+    """
+    Search the python representation of a Json Dict or list for a field
+    whose value matches the given regex. Returns a list of matched dicts.
+    The value will match a string, or any value in a list of strings.
+
+    This function is a wrapper for search_json_field.
+
+    :param js_obj: A dict or list from a parsed json file
+    :param field: The name of the field to be found
+    :param regex_str: regular expression to search values for
+    :return: a list of matching dictionaries.
+    """
+    regex = re.compile(regex_str)
+
+    if isinstance(js_obj, list) or isinstance(js_obj, dict):
+        return [i for i in search_json_field(js_obj, field, regex)
+                if i is not None]
+    else:
+        raise InvalidOM
+
+
+###
+# templates
+###
+def clean_str(a_str: str) -> str:
+    """
+    Clean up a multiline string so that the line begins after the '|'
+    character. Useful for guaranteeing exact format of a long string.
+
+    example:
+    a = '''
+    |this is
+    |  that is
+    |
+    | foo
+    '''
+
+    clean_str(a) => 'this is\n  that is\n\n foo'
+    """
+    def post_pipe(b_str: str):
+        return b_str.split('|', 1)[-1]
+    return "\n".join(post_pipe(l).rstrip() for l in a_str.splitlines())
+
+
+def generate_vtable(device_name: str, reg_list: t.List[Register]) -> str:
+    """
+    Generate the vtable entries for a device and set of register.
```

Comment:
```suggestion
    Generate the vtable entries for a device and set of registers.
```


Can you also add a description of what the vtable entries are for? Also, are these just the prototypes and not the definitions?

Reply:
just declartions. will add to name/comment